### PR TITLE
Add RPCs for RemoteCluster transition to gRPC

### PR DESCRIPTION
**Pull request by strideynet**
Part of https://github.com/gravitational/teleport/issues/38669
Prerequisite of https://github.com/gravitational/teleport/pull/38671

Introduce proto spec for a new Presence gRPC service with the RemoteCluster RPCs.

### CODE REVIEW

File: api/proto/teleport/presence/v1/service.proto [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+package teleport.presence.v1;
+
+import "google/protobuf/empty.proto";
+import "teleport/legacy/types/types.proto";
+
+option go_package = "github.com/gravitational/teleport/api/gen/proto/go/teleport/presence/v1;presencev1";
+
+// PresenceService provides methods to manage presence of RemoteClusters
+service PresenceService {
```
Comment (strideynet) [inline api/proto/teleport/presence/v1/service.proto:25]: I've omitted Upsert as we don't currently have that on the HTTP side, open to arguments that I should include it.

Comment (flyinghermit) [reply api/proto/teleport/presence/v1/service.proto:25]: The resource guideline also suggests to prefer Create and Update over Upsert, so this should be fine? - https://github.com/gravitational/teleport/blob/master/rfd/0153-resource-guidelines.md#upsert

**Review by flyinghermit — APPROVED** [review]

File: api/proto/teleport/presence/v1/service.proto (hunk 1 above)
Comment (strideynet) [reply api/proto/teleport/presence/v1/service.proto:25]: > I don't know that we should be exposing a Create/Update/Upsert for a remote cluster. They are intended to only be created internally by Auth during the validation process of adding a trusted cluster to the leaf. From what I can tell the existing REST API only exposes Create but I can't find any usages of that.

We also expose an `UpdateRemoteCluster` on the big auth server gRPC API - it looks like this is exposed in `tctl`.

Comment (strideynet) [reply api/proto/teleport/presence/v1/service.proto:25]: More than happy to drop the Create RPC from this and just delete the HTTP endpoint/client methods - less work for me!

Comment (strideynet) [reply api/proto/teleport/presence/v1/service.proto:25]: c4e25b1527

File: api/proto/teleport/presence/v1/service.proto [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+  // Token to retrieve the next page of results, or empty if there are no
+  // more results in the list.
+  string next_page_token = 2;
+}
+
+// Request for UpdateRemoteCluster
+message UpdateRemoteClusterRequest {
+  // RemoteCluster is the RemoteCluster to update.
+  types.RemoteClusterV3 remote_cluster = 1;
```
[the remark this answers is not in the document]
Comment (strideynet) [reply api/proto/teleport/presence/v1/service.proto:66]: [42e15f1](https://github.com/gravitational/teleport/pull/38793/commits/42e15f15626f1397097ce8152c5e58b2e6be5f49)

**Review by rosstimothy — APPROVED** [review]

File: api/proto/teleport/presence/v1/service.proto [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+// Request for ListRemoteClusters
+message ListRemoteClustersRequest {
+  // The maximum number of items to return.
+  // The server may impose a different page size at its discretion.
+  int32 page_size = 1;
+
+  // The next_page_token value returned from a previous List request, if any.
+  string page_token = 2;
```
Comment (tigrato) [inline api/proto/teleport/presence/v1/service.proto:50]:
```suggestion
  string next_page_token = 2;
```

**Review by tigrato — APPROVED** [review]
please update the field name before merging

File: api/proto/teleport/presence/v1/service.proto (hunk 3 above)
Comment (strideynet) [reply api/proto/teleport/presence/v1/service.proto:51]: 0abf323118

Comment (tigrato) [reply api/proto/teleport/presence/v1/service.proto:51]: Oh my bad 😞 
I should read everything to the end

**Pull request merged**